Partial revert of upstream commit bfa864e1645e140da2e1aae3cf0d0ba0674f6eb5
until gcc stops providing fix-includes.

--- a/misc/sys/cdefs.h
+++ b/misc/sys/cdefs.h
@@ -412,6 +412,14 @@
 # define __glibc_has_attribute(attr)	0
 #endif
 
+#ifdef __has_include
+/* Do not use a function-like macro, so that __has_include can inhibit
+   macro expansion.  */
+# define __glibc_has_include __has_include
+#else
+# define __glibc_has_include(header)	0
+#endif
+
 #if (!defined _Noreturn \
      && (defined __STDC_VERSION__ ? __STDC_VERSION__ : 0) < 201112 \
      &&  !__GNUC_PREREQ (4,7))
